spec/controllers/users_controller_spec.rb

```diff
@@ -2,11 +2,43 @@
 
 RSpec.describe UsersController, type: :controller do
 
+  before(:each) do
+    request.env["HTTP_REFERER"] = root_url
+    @user = User.new(:email => 'el@el.com', :password => '1234567')
+    @user.save
+    # @user = User.create!(email: "test@test.com", password: 'testtest')
+  end
+
   describe "GET #index" do
     it "returns http success" do
       get :index
       expect(response).to have_http_status(:success)
     end
   end
 
+  describe "GET #show" do
+    it "assign user return as @user" do
+      get :show, params: { id: @user.id }
+    end
+  end
+
+  describe "GET #edit" do
```

Comment:
It should also be possible to test that the controller renders the correct template